x/circuit/module.go

```diff
@@ -113,6 +119,26 @@ func (am AppModule) ExportGenesis(ctx context.Context) (json.RawMessage, error)
 	return am.cdc.MarshalJSON(gs)
 }
 
+// TxValidator implements appmodule.HasTxValidation.
+// It replaces auth ante handlers for server/v2
+func (am AppModule) TxValidator(ctx context.Context, tx transaction.Tx) error {
```

Comment:
this part feels like it should be handled by the wiring, we should not be building the ante handler here

Reply:
For circuit we will have a tx validator and a pre message handler basically doing the same thing, do we actually then need a pre message handler?

Reply:
we should be able to drop it in that sense. but for backwards compatibility we still need this